pkg/lint/linter/config.go

```diff
@@ -133,11 +133,12 @@ func (lc *Config) Name() string {
 	return lc.Linter.Name()
 }
 
-func (lc *Config) WithNoopFallback(cfg *config.Config) *Config {
-	if cfg != nil && config.IsGreaterThanOrEqualGo122(cfg.Run.Go) {
+func (lc *Config) WithNoopFallback(cfg *config.Config, cond func(cfg *config.Config) bool) *Config {
+	if cfg != nil && cond(cfg) {
 		lc.Linter = &Noop{
-			name: lc.Linter.Name(),
-			desc: lc.Linter.Desc(),
+			name:   lc.Linter.Name(),
+			desc:   lc.Linter.Desc(),
+			reason: "This linter is disabled because the Go version of your project is lower than Go 1.22.",
```

Comment:
This field feels very specific to some lintes and not generic enough to always be used when using `WithNoopFallback`. Should we maybe accept a reason as input or just wait until a different reason is needed?

Reply:
When I made this change, I was thinking of the same thing, but for now, it's always the same message.
If we need something different in the future we will do it.